### PR TITLE
Improving opal comment4

### PR DESCRIPTION
**Pull request by Ducasse**
See the commits.
Should check how to introduced a renamed classes IRTranslatorVisitor and IRSimpleOptimizerVisitor

### CODE REVIEW

**Comment by Ducasse** [comment]
First let us see if this one is passing

File: src/OpalCompiler-Core/IRSimpleOptimizerVisitor.class.st [hunk 1]
```diff
@@ -0,0 +1,77 @@
+"
+I perform some simple optimizations by changing IR nodes.
+I transform:
+
+- sequence of store, pop into popInto
+- some returns into quick returns
+"
+Class {
+	#name : 'IRSimpleOptimizerVisitor',
+	#superclass : 'IRVisitor',
```
Comment (guillep) [inline src/OpalCompiler-Core/IRSimpleOptimizerVisitor.class.st:10]: Hi @Ducasse this class is not replacing the old one, is that on purpose?

File: src/OpalCompiler-Core/IRTranslatorVisitor.class.st [hunk 2]
```diff
@@ -0,0 +1,337 @@
+"
+I responsible to translate the IR instructions into bytecode. 
+I delegate, however such a task to a BytecodeGenerator.
+So I visit IRMethod instructions, sending the appropriate messages to my BytecodeGenerator (gen). 
+
+Check `IRMethod>>#generate`.
+"
+Class {
+	#name : 'IRTranslatorVisitor',
+	#superclass : 'IRVisitor',
```
Comment (guillep) [inline src/OpalCompiler-Core/IRTranslatorVisitor.class.st:10]: Hi @Ducasse this class is not replacing the old one, is that on purpose?

**Comment by guillep** [comment]
Ok, we talked and the replacement of users and removal of old classes comes in a separate step.

**Pull request merged**